src/core/namespace.c

```diff
@@ -150,6 +154,14 @@ static const BindMount protect_system_strict_table[] = {
         { "/root",               READWRITE,    true  },      /* ProtectHome= */
 };
 
+/* If RootDirectory= is used, let's mount /sys and /proc into the it, but only as a fallback if the user hasn't mounted
+ * something there already. These mounts are hence overriden by any other explicitly configured mounts. */
+static const BindMount root_directory_table[] = {
+        { "/proc",               PROCFS,       false },
+        { "/dev",                BIND_DEV,     false },
+        { "/sys",                SYSFS,        false },
```

Comment:
@poettering , sorry, I missed a notification.

Hm, I think some people don't want to expose these filesystems. What should they do?